test/rpc_test.py

```diff
@@ -1,24 +1,33 @@
 #!/usr/bin/env python3
 from __future__ import absolute_import, division, print_function, unicode_literals
 
+import functools
 import sys
 import unittest
 from os import getenv
+from unittest import mock
 
 import torch
 import torch.distributed as dist
+import torch.distributed.rpc_backend_registry as rpc_backend_registry
 
 
 if not dist.is_available():
     print("c10d not available, skipping tests")
     sys.exit(0)
 
-from torch.distributed.rpc import RpcBackend
 from common_utils import load_tests
+from torch.distributed.rpc_api import RpcBackend
```

Comment:
I haven't seen this module before, is it new? If so, why was it added?

Reply:
Nevermind, I see it is added here. Why the rename?

Reply:
"rpc" module was shadowed by "rpc" method imported to "torch.distributed".
This caused the patch replacing the wrong target.
Fixed it by rename the module to "rpc_api".

Reply:
I see, thanks. I believe @rohan-varma is working on a PR to rename those to `rpc_sync` and `rpc_async`, right? Then this will no longer be necessary.

Reply:
@pietern @xush6528 , that change is landed in https://github.com/pytorch/pytorch/pull/26570. But it doesn't remove the `rpc` function, it instead adds a warning to callers who use it and encourage them to call `rpc_sync` or `rpc_async`. We eventually want to entirely remove the `rpc` function as you mentioned, but wasn't sure if we could do this yet due to backwards compatibility issues, so after a discussion with @mrshenli (comments on the PR) we decided to go with this approach.

Reply:
@rohan-varma When we do, can you make sure to revert the rename? If it's an internal only module, it shouldn't be named `_api`, but rather use an underscore prefix, like all other private API like things in PyTorch.

Reply:
@pietern, the `rpc_api.py` module is part of our external-facing API though, right? It contains things like `get_worker_id`, `remote, `rpc_sync`, etc. Should we define `__all__` to make it clear what our external facing APIs are?

Reply:
I feel ideally we want to put all external api names in `__init__.py` and has a module implementing these api names. 

It's a mess at the moment.
In `__init__.py` defines `def init_model_parallel`
In `rpc_api.py`, other API is defined (as Rohan listed).

Reply:
@pietern @xush6528 , I've created #27234 to track this and explore possible ideas